app/src/main/java/com/thenewboston/kotlinsdk/home/views/BankOverviewFragment.kt

```diff
@@ -0,0 +1,86 @@
+package com.thenewboston.kotlinsdk.home.views
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import com.thenewboston.kotlinsdk.BuildConfig
+import com.thenewboston.kotlinsdk.GENERAL_ERROR
+import com.thenewboston.kotlinsdk.R
+import com.thenewboston.kotlinsdk.home.viewmodels.BankViewModel
+import com.thenewboston.kotlinsdk.network.models.BankConfigModel
+import kotlinx.android.synthetic.main.bank_overview_fragment.*
+import kotlinx.android.synthetic.main.bank_overview_fragment.errorText
+import kotlinx.android.synthetic.main.bank_overview_fragment.ip_address
+import kotlinx.android.synthetic.main.bank_overview_fragment.node_type
+import kotlinx.android.synthetic.main.bank_overview_fragment.port
+import kotlinx.android.synthetic.main.bank_overview_fragment.protocol
+import kotlinx.android.synthetic.main.bank_overview_fragment.tx_fee
+import kotlinx.android.synthetic.main.bank_overview_fragment.version
+import kotlinx.android.synthetic.main.bank_overview_fragment.wholeLayout
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class BankOverviewFragment : Fragment() {
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        return inflater.inflate(R.layout.bank_overview_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val viewModel = requireActivity().run {
+            ViewModelProvider(this).get(BankViewModel::class.java)
+        }
+
+        active_bank.text = BuildConfig.BANK_BASE_URL
+
+        CoroutineScope(Dispatchers.IO).launch {
+            getConfig(viewModel)
+            getNumConfirmationServices(viewModel)
+        }
+    }
+
+    private suspend fun getConfig(viewModel: BankViewModel) {
+        val data = viewModel.getBankConfig()
+        withContext(Dispatchers.Main) {
```

Comment:
Pro tip: Never make network calls inside the view i.e. fragments or activity. The View Model should be responsible for making the request and sending it back to the view.

Reply:
Yeah the actual network call is done in repository here, I am calling a function inside viewmodel, that executes the network call in the repo by calling the function in repo and returns the result. Here I am triggering that viewmodel function from view.

Reply:
Ok have you used LiveData before? In the ViewModel you can use LiveData to observe for the result and send it to the view.

Reply:
Yeah here I am using mutable live data to keep track of the account number, when it's changed, data related to acc no is replaced. For the rest, we were doing network calls everytime the tab is opened in the account manager electron app so kept it that way here as well.

Reply:
Ok when you make the network request how do you cancel the coroutine scope?

Reply:
Yeah I understood the issue, I've started integrating livedata for the rest of the modules and making network requests from viewmodel today like you recommended. 🌞